src/api.js

```diff
@@ -17,33 +18,376 @@ const getMessages = args => {
+      for (const name of composition.messages.map(message => message.actorName)) {
+        actorName += name + ' | ';
+      }
+      composition.actorName = actorName.substring(0, actorName.length - 3);
+    }
+    // add status
+    const statuses = composition.messages.map(message => message.status);
+    if (statuses.includes('Failure')) {
+      composition.status = 'Failure';
+    } else if (statuses.includes('Skipped')) {
+      composition.status = 'Skipped';
+    } else if (statuses.includes('Canceled')) {
+      composition.status = 'Canceled';
+    } else if (statuses.includes('Started')) {
+      composition.status = 'Started';
+    } else if (statuses.every(status => status === 'Success')) {
+      composition.status = 'Success';
+    } else if (statuses.every(status => status === 'Not yet enqueued')) {
+      composition.status = 'Not yet enqueued';
+    } else if (
+      (composition.type === 'group' && statuses.every(status => status === 'Pending')) ||
+      (composition.type === 'pipeline' && statuses[0] === 'Pending')
+    ) {
+      composition.status = 'Pending';
+    } else {
+      composition.status = 'Started';
+    }
+    // add Priority
+    composition.priority = composition.messages[0].priority;
+    // add Started time
+    if (composition.type === 'pipeline') {
+      composition.startedDatetime = composition.messages[0].startedDatetime;
+    } else {
+      const datetime = Math.min(
+        ...composition.messages.map(message =>
+          message.startedDatetime ? message.startedDatetime.valueOf() : Infinity
+        )
+      );
+      if (datetime !== Infinity) {
+        composition.startedDatetime = new Date(datetime);
+      }
+    }
+    // add Wait time
+    let wait_time = 0;
+    composition.messages.forEach(message => {
+      if (message.waitTime) {
+        wait_time += message.waitTime;
+      }
+    });
+    composition.waitTime = wait_time;
+    // add Execution time
+    let execution_time = 0;
+    composition.messages.forEach(message => {
+      if (message.executionTime) {
+        execution_time += message.executionTime;
+      }
+    });
+    composition.executionTime = execution_time;
+    //add Remaining Time
+    if (composition.messages.every(message => message.endDatetime || message.remainingTime)) {
+      let remaining_time = 0;
+      composition.messages.forEach(message => {
+        if (message.remainingTime) {
+          remaining_time = Math.max(remaining_time, message.remainingTime);
+        }
+      });
+      composition.remainingTime = remaining_time;
+    }
+    // add Progress
+    if (
+      composition.messages.every(message => message.status === 'Success' || message.progress) &&
+      !composition.messages.every(message => message.status === 'Success')
+    ) {
+      composition.progress =
+        composition.messages
+          .map(message => message.progress || 1)
+          .reduce((value, acc) => value + acc) / composition.messages.length;
```

Comment:
value and acc are inversed here. I would merge with the map 
```suggestion
          .reduce((acc, message) => ACC + (message.progress || 1)) / composition.messages.length;
```